n_plus_one_example/src/main.rs

```diff
@@ -0,0 +1,30 @@
+pub mod db;
+use color_eyre::eyre::Result;
+use db::{connect_db, fetch_posts};
+use std::collections::HashMap;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let (client, _connection_handle) = connect_db().await?;
+
+    let comment_counts = comment_count_per_post(&client).await?;
+    println!("Fetched comment counts for {} posts.", comment_counts.len());
+
+    for (post_id, count) in comment_counts {
+        println!("Post ID {} has {} comments.", post_id, count);
```

Comment:
we could swap variables. However I know you will not be the biggest fan of this. Just throwing it out there 😁